activerecord/lib/active_record/migration.rb

```diff
@@ -127,7 +128,13 @@ def initialize(name = nil)
     end
   end
 
-  class PendingMigrationError < MigrationError#:nodoc:
+  class PendingMigrationError < MigrationError #:nodoc:
+    include ActiveSupport::ActionableError
+
+    action "Run pending migrations" do
+      ActiveRecord::Tasks::DatabaseTasks.migrate
```

Comment:
I don't think this location should know about the DatabaseTasks. How about we let action accept a command option instead? E.g.:

```ruby
action "Run pending migrations", command: "db:migrate"

def action(name, command: nil, &block)
  _actions[name] = block || -> { Rails::Command.invoke command }
end
```

Although then Active Support knows about Rails commands… 😄

Reply:
Let's deal with this part later. I do like the idea that other libraries could extend `action` with their own behaviors ala:

```ruby
# Somewhere in railties
ActiveSupport::ActionableErrors.register_behavior :command do |command|
  Rails::Command.invoke command
end

# Back in Active Support:
def action(name, **behaviors, &block)
  _actions[name] = block || extract_behavior(behaviors)
end

def extract_behavior(behaviors)
  if match = (behaviors.keys & self.class.behaviors.keys).first
    callable, value = self.class.behaviors[match], behaviors[match]

    -> { callable.call(value) }
end
```

Hm, but then we'll have load order dependencies… 😅 Well, this was fun to think about! I still think we should find a way such that Migration doesn't know about the `migrate` task, but that's later.

Reply:
Hey, yeah, I like that we can execute arbitrary code here now and we're not explicitly coupled with commands, but they are kinda easy to invoke if that's your desire.